wagtail/users/views/users.py

```diff
@@ -70,74 +70,70 @@ def get_users_filter_query(q, model_fields):
     return conditions
 
 
-@any_permission_required(add_user_perm, change_user_perm, delete_user_perm)
-@vary_on_headers("X-Requested-With")
-def index(request, *args):
-    q = None
-    is_searching = False
-
-    group = None
-    group_filter = Q()
-    if args:
-        group = get_object_or_404(Group, id=args[0])
-        group_filter = Q(groups=group) if args else Q()
+class Index(IndexView):
+
+    any_permission_required = ["add", "change", "delete"]
+    permission_policy = ModelPermissionPolicy(User)
+    model = User
+    context_object_name = "users"
+    index_url_name = "wagtailusers_users:index"
+    add_url_name = "wagtailusers_users:add"
+    edit_url_name = "wagtailusers_users:edit"
+    default_ordering = "name"
+    paginate_by = 20
+    template_name = None
+    is_searchable = True
+    page_title = gettext_lazy("Users")
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        setattr(self, "template_name", self.get_template())
+        self.group = get_object_or_404(Group, id=args[0]) if args else None
+        self.group_filter = Q(groups=self.group) if self.group else Q()
+        self.model_fields = [f.name for f in User._meta.get_fields()]
+
+    def get_queryset(self):
+        if self.is_searching:
+            conditions = get_users_filter_query(self.search_query, self.model_fields)
+            users = User.objects.filter(self.group_filter & conditions)
+        else:
+            users = User.objects.filter(self.group_filter)
 
-    model_fields = [f.name for f in User._meta.get_fields()]
+        if self.locale:
+            users = users.filter(locale=self.locale)
 
-    if "q" in request.GET:
-        form = SearchForm(request.GET, placeholder=_("Search users"))
-        if form.is_valid():
-            q = form.cleaned_data["q"]
-            is_searching = True
-            conditions = get_users_filter_query(q, model_fields)
+        if "last_name" in self.model_fields and "first_name" in self.model_fields:
+            users = users.order_by("last_name", "first_name")
 
-            users = User.objects.filter(group_filter & conditions)
-    else:
-        form = SearchForm(placeholder=_("Search users"))
+        if self.get_ordering() == "username":
+            users = users.order_by(User.USERNAME_FIELD)
 
-    if not is_searching:
-        users = User.objects.filter(group_filter)
+        return users
 
-    if "last_name" in model_fields and "first_name" in model_fields:
-        users = users.order_by("last_name", "first_name")
+    @property
+    def requested_with_XMLHttpRequest(self):
+        return self.request.headers.get("x-requested-with") == "XMLHttpRequest"
 
-    if "ordering" in request.GET:
-        ordering = request.GET["ordering"]
+    def get_template(self):
+        if self.requested_with_XMLHttpRequest:
```

Comment:
```suggestion
        if self.request.headers.get("x-requested-with") == "XMLHttpRequest":
```
As above - let's go with the check in-line here - similar to other code like this - wagtail/models/__init__.py & wagtail/users/views/groups.py